packages/cdktf-cli/lib/dependencies/prebuilt-providers.ts

```diff
@@ -21,15 +21,68 @@ type ProvidersMap = {
   [name: string]: string;
 };
 
+const fetchCache = new Map<string, any>();
```

Comment:
Nice! This closes https://github.com/hashicorp/terraform-cdk/issues/1815 already 😁